static/src/javascripts/projects/common/modules/email/email-article.js

```diff
@@ -0,0 +1,308 @@
+// @flow
+import $ from 'lib/$';
+import bean from 'bean';
+import bonzo from 'bonzo';
+import fastdom from 'fastdom';
+import email from 'common/modules/email/email';
+import config from 'lib/config';
+import iframeTemplate from 'raw-loader!common/views/email/iframe.html';
+import template from 'lodash/utilities/template';
+import { spaceFiller } from 'common/modules/article/space-filler';
+import { logError } from 'lib/robust';
+import emailRunChecks from 'common/modules/email/run-checks';
+import { session } from 'lib/storage';
+import { trackNonClickInteraction } from 'common/modules/analytics/google';
+import { waitForCheck } from 'common/modules/check-mediator';
+
+import type { SpacefinderRules } from 'common/modules/spacefinder.js';
+
+const insertBottomOfArticle = ($iframeEl: bonzo): void => {
+    fastdom.write(() => $iframeEl.prependTo('.content-footer'));
+};
+
+type ListConfig = {
+    listId: string,
+    listName: string,
+    campaignCode: string,
+    displayName: {
+        normalText: string,
+        accentedText: string,
+    },
+    headline: string,
+    description: string,
+    successHeadline: string,
+    successDescription: string,
+    insertMethod: Function,
```

Comment:
What do you think about being super-specific with this type:
```js
insertMethod: (bonzo) => void
```
Flow docs tell us [to avoid the `Function` type](https://flow.org/en/docs/types/functions/#toc-function-type) as it's just a fancy way of opting out of typing.

Reply:
oh nice one